pkg/etcdmgr/bootstrap/discovery.go

```diff
@@ -70,32 +92,79 @@ func queryDiscoveryService(token string) (*store.Event, error) {
 	return &res, nil
 }
 
-func GetCurrentNodesFromDiscovery(token string) ([]string, error) {
+// Get the nodes that have registered with the discovery service.
+// We only want to return the number of nodes that are expected for the etcd cluster size.
+// Etcd will not allow more etcd servers to start unless the AddMember api is called.
+// Here we will ignore any nodes that have registered beyond the expected size.
+// It is important to return the nodes with the lowest index for consistent behavior in the cluster.
+func GetCurrentNodesFromDiscovery(token string, size int) ([]string, error) {
 	res, err := queryDiscoveryService(token)
 	if err != nil {
 		return nil, err
 	}
 
-	nodes := make([]string, 0, len(res.Node.Nodes))
+	var upperIndex uint64
+	var ignored []string
+	nodeMap := map[uint64][]string{}
 	for _, nn := range res.Node.Nodes {
 		if nn.Value == nil {
 			logger.Debugf("Skipping %q because no value exists", nn.Key)
 		}
 
-		n, err := newDiscoveryNode(*nn.Value, DefaultClientPort)
+		endpoints, err := newDiscoveryNode(*nn.Value, DefaultClientPort)
 		if err != nil {
 			logger.Warningf("invalid peer url %q in discovery service: %v", *nn.Value, err)
 			continue
 		}
 
-		for _, node := range n {
+		upperIndex, ignored = addNodeToMap(nodeMap, endpoints, size, nn.CreatedIndex, upperIndex, ignored)
+	}
+
+	if len(ignored) > 0 {
+		logger.Infof("Ignored extra etcd members: %+v", ignored)
```

Comment:
i think it works well as info. It's an expected (or at least handled) condition rather than something to worry about